src/dmd/root/file.d

```diff
@@ -22,6 +22,320 @@ import dmd.root.filename;
 import dmd.root.rmem;
 import dmd.root.string;
 
+/**
+Runs a non-pure function or delegate as pure code. Use with caution.
+Returns: whatever `fun` returns.
```

Comment:
Needs `Params:` section.

Please move private leaf functions to the end of the file. Public APIs should come first.

Reply:
done